builder/proxmox/common/step_start_vm.go

```diff
@@ -318,36 +359,51 @@ func generateProxmoxDisks(disks []diskConfig) *proxmox.QemuStorages {
 					EmulateSSD:      disks[idx].SSD,
 				},
 			}
-			// We need reflection here as the storage objects are not exposed
-			// as a slice, but as a series of named fields in the structure
-			// that the APIs use.
-			//
-			// This means that assigning the disks in the order they're defined
-			// in would result in a bunch of `switch` cases for the index, and
-			// named field assignation for each.
-			//
-			// Example:
-			// ```
-			// switch ideCount {
-			// case 0:
-			//	dev.Disk_0 = dev
-			// case 1:
-			//	dev.Disk_1 = dev
-			// [...]
-			// }
-			// ```
-			//
-			// Instead, we use reflection to address the fields algorithmically,
-			// so we don't need to write this verbose code.
-			reflect.
-				// We need to get the pointer to the structure so we can
-				// assign a value to the disk
-				ValueOf(&ideDisks).Elem().
-				// Get the field from its name, each disk's field has a
-				// similar format 'Disk_%d'
-				FieldByName(fmt.Sprintf("Disk_%d", ideCount)).
-				Set(reflect.ValueOf(&dev))
-			ideCount++
+			for {
```

Comment:
Bit worried about this infinite loop, is there a chance we can never leave here? At first glance this looks risky but maybe this is fine.

Reply:
Good point, I was relying on the iso/additional_iso_files/disk validation from the builder's Prepare function to avoid overallocation. 
`QemuIdeDisks{} QemuSataDisks{} QemuScsiDisks{} QemuVirtIODisks{}` from the SDK have a fixed number of fields as per the limitations of each bus type in Proxmox, I imagine the loop would panic when it hits a Disk_ field that doesn't exist.
I've added a check at the start of the loop to break when all struct fields are exhausted.

Reply:
More than a panic I was worried the plugin could be stuck in this loop, which is not trivial to debug when it happens in the wild.
Panics are fine imho, especially if the panic occurs because of a logic bug from the `Prepare`, we can investigate and fix them then. Silent continuations like those introduced with that reroll are more insidious in my experience, as users barely know why the plugin tells them `Max 4 devices` in the verbose logs and doesn't error, but they'll wonder why their device doesn't show up in their configs.
I would maybe suggest returning an error here if we bust that limit, panicking might be suboptimal because of the stacktrace that is returned (although for debugging this is kinda good, and it explicitly tells users to report the problem), and silently continuing feels wrong to me.

Reply:
Yeah I made the wrong call here 😅 I can update this to be a ui response rather than a log.Print